src/ast/nodes/ChainExpression.ts

```diff
@@ -1,9 +1,50 @@
+import type MagicString from 'magic-string';
+import type { RenderOptions } from '../../utils/renderHelpers';
+import type { DeoptimizableEntity } from '../DeoptimizableEntity';
+import type { HasEffectsContext } from '../ExecutionContext';
+import { SHARED_RECURSION_TRACKER } from '../utils/PathTracker';
+import { EMPTY_PATH } from '../utils/PathTracker';
 import type CallExpression from './CallExpression';
 import type MemberExpression from './MemberExpression';
 import type * as NodeType from './NodeType';
+import type { LiteralValueOrUnknown } from './shared/Expression';
+import { UnknownValue } from './shared/Expression';
 import { NodeBase } from './shared/Node';
 
-export default class ChainExpression extends NodeBase {
+const unset = Symbol('unset');
+
+export default class ChainExpression extends NodeBase implements DeoptimizableEntity {
 	declare expression: CallExpression | MemberExpression;
 	declare type: NodeType.tChainExpression;
+	private objectValue: LiteralValueOrUnknown | typeof unset = unset;
+
+	deoptimizeCache(): void {
+		this.objectValue = UnknownValue;
```

Comment:
This one is actually hard to cover in a test and I am fine if it is not covered. To hit this, you have to use a variable further up in the chain that is initialized as `undefined` but later gets another value assigned, invalidating the `object.getLiteralValueAtPath` in line 46.
The problem is to find a suitable "later". Sometimes it works to do this in a function but as I said, it is tricky.